doc/weakly_supervised.rst

```diff
@@ -196,8 +228,60 @@ programming.
 LSML
 ----
 
-`LSML`: Metric Learning from Relative Comparisons by Minimizing Squared
-Residual
+Metric Learning from Relative Comparisons by Minimizing Squared Residual
+(:py:class:`LSML <metric_learn.lsml.LSML>`)
+
+`LSML` proposes a simple, yet effective, algorithm that minimizes a convex 
+objective function corresponding to the sum of squared residuals of 
+constraints. This algorithm uses the constraints in the form of the 
+relative distance comparisons, such method is especially useful where 
+pairwise constraints are not natural to obtain, thus pairwise constraints 
+based algorithms become infeasible to be deployed. Furthermore, its sparsity 
+extension leads to more stable estimation when the dimension is high and 
+only a small amount of constraints is given.
+
+The loss function of each constraint 
+:math:`d(\mathbf{x}_a, \mathbf{x}_b) < d(\mathbf{x}_c, \mathbf{x}_d)` is 
+denoted as:
+
+.. math::
+
+    L(d(\mathbf{x}_a, \mathbf{x}_b) < d(\mathbf{x}_c, \mathbf{x}_d)) = 
```

Comment:
We could get rid of the notation L, which is redundant with H. You could define directly H(d_\mathbf{M}(\mathbf{x}_a, \mathbf{x}_b), which is what you use in the final problem formulation. Of course you can rename H by L then ;-)

Reply:
done

Reply:
I think you have forgotten this one

Reply:
nevermind, got confused